testutil/tx/eth.go

```diff
@@ -59,7 +59,13 @@ func PrepareEthTx(
 		msg.From = ""
 
 		txGasLimit += msg.GetGas()
-		txFee = txFee.Add(sdk.Coin{Denom: utils.BaseDenom, Amount: sdkmath.NewIntFromBigInt(msg.GetFee())})
+		// NOTE: This is not always going be true, but we assume 6 decimals
+		// for the Evmos chain is always going to be uatom
+		if evmParams.DenomDecimals == types.Denom6Dec {
+			txFee = txFee.Add(sdk.Coin{Denom: evmParams.EvmDenom, Amount: sdkmath.NewIntFromBigInt(msg.GetFee())})
+		} else {
+			txFee = txFee.Add(sdk.Coin{Denom: evmParams.EvmDenom, Amount: sdkmath.NewIntFromBigInt(msg.GetFee())})
+		}
```

Comment:
Both of these are the same. Should we convert one to 6 dec?

Reply:
Ohh true the change was to just get it from evm params, the conversion is done later down the stack.